models/User.js

```diff
@@ -100,5 +100,17 @@ module.exports = (sequelize, Sequelize) => {
       onDelete: "CASCADE",
     });
   };
+  // Anonymize comments when a user is soft-deleted:
+  User.addHook("afterDestroy", async (user, options) => {
+    await sequelize.models.Comment.update(
+      {
+        isDeleted: true,
+      },
+      {
+        where: { userId: user.id },
+        transaction: options.transaction,
+      }
+    );
+  });
```

Comment:
This doesn't seem to be working, with our user delete function as it stands. Apparently this bypasses the hook
```js
  async softDelete(id) {
    return this.User.destroy({ where: { id } });
  }
```
The user delete would need to be set to something like this
```js
  async softDelete(id) {
    const user = await this.User.findByPk(id);
    return user.destroy();
  }
  ```
  
  Using a where in a destroy function on the user model is consider bulk and hence the hook doesn't trigger. We need to find by using findOne, or findByPk then use the destroy helper function attached to it to trigger the hook.